cmd/minishift/cmd/start_preflight.go

```diff
@@ -364,12 +370,38 @@ func checkLibvirtDefaultNetworkActive() bool {
 
 // checkHypervDriverSwitch returns true if Virtual Switch has been selected
 func checkHypervDriverSwitch() bool {
-	switchEnv := os.Getenv("HYPERV_VIRTUAL_SWITCH")
-	if switchEnv == "" {
+	switchEnvName := "HYPERV_VIRTUAL_SWITCH"
+	posh := powershell.New()
+	defer posh.Close()
+
+	switchEnvValue := os.Getenv(switchEnvName)
+
+	if switchEnvValue == "" {
+		checkIfDefaultSwitchExists := `Get-VMSwitch -Id c08cb7b8-9b3c-408e-8e30-5e16a3aeb444 | ForEach-Object { $_.SwitchType }`
+		fmt.Printf("\n   'Default Switch' ... ")
+		stdOut, _ := posh.Execute(checkIfDefaultSwitchExists)
+		if strings.Contains(stdOut, "Internal") {
+
+			// force setting the environment variable
+			os.Setenv("HYPERV_VIRTUAL_SWITCH", "Default Switch")
+			return true
+		}
+
 		return false
 	}
 
-	return true
+	if switchEnvValue != "" {
+		checkIfVirtualSwitchExists := fmt.Sprintf("Get-VMSwitch %s| ForEach-Object { $_.SwitchType }", switchEnvValue)
+		fmt.Printf(fmt.Sprintf("\n   '%s' ... ", switchEnvValue))
+		stdOut, _ := posh.Execute(checkIfVirtualSwitchExists)
+		if strings.Contains(stdOut, "Get-VMSwitch") {
+			return false // error returned
```

Comment:
This comment is odd :)

Reply:
#2096 will remove this likely...

... also, the get-vmswitch test needs to be made case-insensitive... filing an issue upstream